paddle/fluid/platform/enforce.h

```diff
@@ -439,6 +439,26 @@ struct EnforceNotMet : public std::exception {
                                         __ROLE, __NAME, __OP_TYPE));        \
   } while (0)
 
+/*
+ * Summary: This macro is used to check whether the op is on CPUPlace
+ * it is used in many places, so abstract this macro.
+ *
+ * Parameters:
+ *     __CTX_GETPLACE: (platform::Place), ctx.GetPlace()
+ *     __OP_TYPE: (string), the op type
+ *
+ * Examples:
+ *    GET_PLACE_CPU_CHECK(ctx.GetPlace(), "Mul");
+*/
+#define GET_PLACE_CPU_CHECK(__CTX_GETPLACE, __OP_TYPE)              \
+  do {                                                              \
+    PADDLE_ENFORCE_EQ(platform::is_cpu_place(__CTX_GETPLACE), true, \
+                      paddle::platform::errors::PreconditionNotMet( \
+                          "This operator %s must run "              \
+                          "CPUPlace",                               \
+                          __OP_TYPE));                              \
+  } while (0)
```

Comment:
What about getting the context as the first argument of the macro?
Also, with a little change in the message.
Like this:
```c++
#define GET_PLACE_CPU_CHECK(__CTX, __OP_TYPE)              \
  do {                                                              \
    PADDLE_ENFORCE_EQ(platform::is_cpu_place((__CTX).GetPlace()), true, \
                      paddle::platform::errors::PreconditionNotMet( \
                          "Operator %s must use CPUPlace",                               \
                          __OP_TYPE));                              \
  } while (0)
```

Reply:
I m worried if we do not what is passed into the function, directly do `.GetPlace()` maybe not safe? 
Anyway from yesterday's meeting, Luotao said I need to separate this PR. Because adding macro need more Baidu's discussion. So I will do this now

Reply:
@wojtuss  I was asked to not use macro in this PR, could you please review again? Thanks